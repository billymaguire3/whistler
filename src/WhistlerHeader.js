import React from 'react';

const WhistlerHeader = () => {
  return (
    <div>
      <h1 className="App-header">
        Whistler: The More Quiet Twitter
      </h1>
    </div>
  );
}

export default WhistlerHeader;
