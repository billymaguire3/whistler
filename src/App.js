import WhistlerHeader from './WhistlerHeader';
import WhistleForm from './WhistleForm';
import React, { Component } from 'react';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      whistles: []
    }
    this.addWhistle = this.addWhistle.bind(this);
  }

  addWhistle(newWhistle) {
    this.setState({
      whistles: [...this.state.whistles, newWhistle]
    })
    return;
  }

  render() {
    return (
      <div className="App">
        <WhistlerHeader />
        <WhistleForm
          addWhistle={this.addWhistle}
        />
      </div>
    );
  }
}

export default App;
