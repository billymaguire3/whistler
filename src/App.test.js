import { render, screen } from '@testing-library/react';
import App from './App';
import WhistlerHeader from './WhistlerHeader';

// Enzyme Config
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Whistler Header', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<WhistlerHeader />);
  })

  test('renders the title of the project', () => {
    expect(wrapper.find('h1').text()).toContain('Whistler: The More Quiet Twitter')
  });
})

describe('Whistle Form', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(<App />);
  })

  test('App renders a form', () => {
    expect(wrapper.find('.whistle-form').exists()).toBeTruthy();
  })

  test('Form renders a label field with correct text', () => {
    expect(wrapper.find('label').text()).toBe('New Whistle Here');
  })

  test('Form renders a text area for new whistles', () => {
    expect(wrapper.find('text-area').exists()).toBeTruthy();
  })

})

