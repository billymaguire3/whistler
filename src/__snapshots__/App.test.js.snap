// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whistle Form Form's onChange method functions correctly 1`] = `ReactWrapper {}`;

exports[`Whistle Form Submission of WhistleForm adds new whistle to the whistles array in state of App 1`] = `ReactWrapper {}`;
